src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -194,8 +194,17 @@ const std::shared_ptr<const Table>& SQLPipelineStatement::get_result_table() {
                 reinterpret_cast<uintptr_t>(this));
   CurrentScheduler::schedule_and_wait_for_tasks(tasks);
 
+  if (_transaction_context && _transaction_context->aborted()) {
+    _mark_as_failed();
+    return _result_table;
+  }
+
   if (_auto_commit) {
-    _transaction_context->commit();
+    const auto success = _transaction_context->commit();
+    if (!success) {
+      _mark_as_failed();
+      return _result_table;
```

Comment:
This means `return nullptr`, correct? Rather be explicit.Couple of lines above as well.

Reply:
Yes, that's correct. But in order to change this we could not return  `shared_ptr<Table>&` anymore as it cannot be null. Would it be better then to return a `shared_ptr<Table>` instead?

Reply:
Got it. Maybe add a comment `// de facto return nullptr, but as a reference`?

Reply:
Or what about `return _result_table = nullptr`?

Reply:
Rather do only one thing per line. Also, readers who have not seen our discussion will have a hard time understanding why this is done.